lib/features/mailbox/data/network/mailbox_api.dart

```diff
@@ -499,4 +502,51 @@ class MailboxAPI with HandleSetErrorMixin {
       throw SetMailboxMethodException(mapErrors);
     }
   }
+
+  Future<GetMailboxByRoleResponse> getMailboxByRole(
+    Session session,
+    AccountId accountId,
+    Role role,
+    {
+      UnsignedInt? limit
+    }
+  ) async {
+    const int defaultLimit = 1;
+    final processingInvocation = ProcessingInvocation();
+    final requestBuilder = JmapRequestBuilder(httpClient, processingInvocation);
+    final mailboxQueryMethod = QueryMailboxMethod(accountId)..addLimit(limit ?? UnsignedInt(defaultLimit));
+
+    final mailboxFilterCondition = MailboxFilterCondition(role: role);
+    mailboxQueryMethod.addFilters(mailboxFilterCondition);
+
+    final mailboxQueryMethodInvocation = requestBuilder.invocation(mailboxQueryMethod);
+    final getMailBoxMethod = GetMailboxMethod(accountId)
+        ..addReferenceIds(processingInvocation.createResultReference(
+          mailboxQueryMethodInvocation.methodCallId,
+          ReferencePath.idsPath,
+        ));
+    final getMailboxInvocation = requestBuilder.invocation(getMailBoxMethod);
+
+    final capabilities = getMailBoxMethod.requiredCapabilities
+      .toCapabilitiesSupportTeamMailboxes(session, accountId);
+
+    final result = await (requestBuilder
+            ..usings(capabilities))
+          .build()
+          .execute();
+    
+    final mailboxResponse = result.parse<GetMailboxResponse>(
+      getMailboxInvocation.methodCallId,
+      GetMailboxResponse.deserialize
+    );
+
+    if (mailboxResponse?.list.isNotEmpty == true) {
+      return GetMailboxByRoleResponse(
+        mailbox: mailboxResponse!.list.first,
+        mailboxState: mailboxResponse.state
```

Comment:
IMO, we don't need it. If you want refresh changes mailbox , you can use the mailbox's `state_cache`. Use `getMailboxState` in `MailboxRepository`